coconut/command/utils.py

```diff
@@ -0,0 +1,174 @@
+            self.fixpickle()
+
+    def fixpickle(self):
+        """Fixes pickling of Coconut header objects."""
+        from coconut import __coconut__
+        for var in self.vars:
+            if not var.startswith("__") and var in dir(__coconut__):
+                self.vars[var] = getattr(__coconut__, var)
+
+    def run(self, code, err=False, run_func=None):
+        """Executes Python code."""
+        try:
+            if run_func is None:
+                exec(code, self.vars)
+            else:
+                return run_func(code, self.vars)
+        except (Exception, KeyboardInterrupt):
+            if err:
+                raise
+            else:
+                traceback.print_exc()
+        except SystemExit:
+            if self.exit is None:
+                raise
+            else:
+                self.exit()
+
+class Console(object):
+    """Manages printing and reading data to the console."""
+    color_code = None
+    on = True
+
+    def __init__(self, main_sig="", debug_sig=""):
+        """Creates the console."""
+        self.main_sig, self.debug_sig = main_sig, debug_sig
+
+    def set_color(self, color=None):
+        """Set output color."""
+        if color:
+            color = color.replace("_", "")
+            if color in color_codes:
+                self.color_code = color_codes[color]
+            else:
+                try:
+                    color = int(color)
+                except ValueError:
+                    raise CoconutException('unrecognized color "'+color+'" (enter a valid color name or code)')
+                else:
+                    if 0 < color <= 256:
+                        self.color_code = color
+                    else:
+                        raise CoconutException('color code '+str(color)+' out of range (must obey 0 < color code <= 256)')
+        else:
+            self.color_code = None
+
+    def add_color(self, inputstring):
+        """Adds the specified color to the string."""
+        if self.color_code is None:
+            return inputstring
+        else:
+            return escape_color(self.color_code) + inputstring + escape_color(end_color_code)
+
+    def display(self, messages, sig="", debug=False):
+        """Prints messages."""
+        if self.on:
+            message = " ".join(str(msg) for msg in messages)
+            for line in message.splitlines():
+                msg = self.add_color(sig + line)
+                if debug is True:
+                    printerr(msg)
+                else:
+                    print(msg)
+
+    def print(self, *messages):
+        """Prints messages with color."""
+        self.display(messages)
+
+    def printerr(self, *messages):
+        """Prints error messages with color and debug signature."""
+        self.display(messages, self.debug_sig, True)
+
+    def show(self, *messages):
+        """Prints messages with color and main signature."""
+        self.display(messages, self.main_sig)
```

Comment:
Please make sure all files end with a newline. In Sublime, this can be automated with `"ensure_newline_at_eof_on_save": true`.